addons/binding/org.openhab.binding.solarview/src/main/java/org/openhab/binding/solarview/handler/solarviewHandler.java

```diff
@@ -0,0 +1,417 @@
+
+    private static final String SOLARVIEW_RESPONSE_ITEM_DELIM = ",";
+    /**
+     * All available channels
+     */
+    private static final String[] channels = { CHANNEL_KT0, CHANNEL_KYR, CHANNEL_KMT, CHANNEL_KDY, CHANNEL_PAC,
+            CHANNEL_UDC, CHANNEL_IDC, CHANNEL_UDCB, CHANNEL_IDCB, CHANNEL_UDCC, CHANNEL_IDCC, CHANNEL_UL1, CHANNEL_IL1,
+            CHANNEL_TKK };
+
+    private final Logger logger = LoggerFactory.getLogger(solarviewHandler.class);
+
+    /** Solarview host */
+    private String hostname;
+    /** refresh interval */
+    private BigDecimal refreshInterval;
+    /** Solarview query string */
+    private String queryString;
+
+    ScheduledFuture<?> refreshJob;
+    /** Energy information */
+    private String energyInformation = null;
+
+    public solarviewHandler(Thing thing) {
+        super(thing);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void initialize() {
+        logger.info("Initializing Solarview handler for '{}'.", getThing().getUID().toString());
+
+        //
+        // Determine query string
+        //
+        ThingTypeUID thingTypeUID = getThing().getThingTypeUID();
+        if (thingTypeUID.equals(THING_TYPE_ENERGY_PRODUCTION)) {
+            this.queryString = SOLARVIEW_REQUEST_ENERGY_PRODUCTION;
+        } else if (thingTypeUID.equals(THING_TYPE_PRODUCTION_INVERTER_ONE)) {
+            this.queryString = SOLARVIEW_REQUEST_PRODUCTION_INVERTER_ONE;
+        } else if (thingTypeUID.equals(THING_TYPE_PRODUCTION_INVERTER_TWO)) {
+            this.queryString = SOLARVIEW_REQUEST_PRODUCTION_INVERTER_TWO;
+        } else if (thingTypeUID.equals(THING_TYPE_PRODUCTION_INVERTER_THREE)) {
+            this.queryString = SOLARVIEW_REQUEST_PRODUCTION_INVERTER_THREE;
+        } else if (thingTypeUID.equals(THING_TYPE_ENERGY_INJECTION)) {
+            this.queryString = SOLARVIEW_REQUEST_ENERGY_INJECTION;
+        } else if (thingTypeUID.equals(THING_TYPE_ENERGY_IMPORT)) {
+            this.queryString = SOLARVIEW_REQUEST_ENERGY_IMPORT;
+        }
+        logger.trace("initialize(): Solarview Query String set to {}.", queryString);
+
+        Configuration config = getThing().getConfiguration();
+        //
+        // hostname parameter
+        //
+        try {
+            hostname = (String) config.get(PROPERTY_SOLARVIEW_SERVER);
+        } catch (Exception e) {
+            logger.warn("Cannot set hostname parameter.", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Can not access Solarview as hostname is invalid.");
+        }
+
+        if (hostname == null) {
+            // let's go for the default
+            hostname = new String(DEFAULTS_SERVERNAME);
+        }
+        //
+        // refreshInterval parameter
+        //
+        try {
+            refreshInterval = (BigDecimal) config.get(PROPERTY_SOLARVIEW_REFRESH_INTERVAL);
+        } catch (Exception e) {
```

Comment:
Fixed by explicitely handling of java.lang.NumberFormatException (caused by invalid user-defined configuration settings).